pythclient/hermes.py

```diff
@@ -0,0 +1,173 @@
+import asyncio
+from typing import TypedDict
+
+import httpx
+import os
+
+from .price_feeds import Price
+
+HERMES_ENDPOINT_HTTPS = "https://hermes.pyth.network/api/"
+HERMES_ENDPOINT_WSS = "wss://hermes.pyth.network/ws"
+
+
+class PriceFeed(TypedDict):
+    feed_id: str
+    price: Price
+    ema_price: Price
+    vaa: str
+
+
+
+class HermesClient:
+    def __init__(self, feed_ids: list[str], endpoint=HERMES_ENDPOINT_HTTPS, ws_endpoint=HERMES_ENDPOINT_WSS):
+        self.feed_ids = feed_ids
+        self.pending_feed_ids = feed_ids
+        self.prices_dict: dict[str, PriceFeed] = {}
+        self.client = httpx.AsyncClient()
+        self.endpoint = endpoint
+        self.ws_endpoint = ws_endpoint
+
+    async def get_price_feed_ids(self) -> list[str]:
+        """
+        Queries the Hermes https endpoint for a list of the IDs of all Pyth price feeds.
+        """
+
+        url = os.path.join(self.endpoint, "price_feed_ids")
+        
+        client = httpx.AsyncClient()
+
+        data = (await client.get(url)).json()
```

Comment:
line 37 seems unnecessary, you can just do `data = (await self.client.get(url)).json()`, generally consider using a single client instance for all HTTP requests within `HermesClient` to utilize connection pooling and reduce overhead -- one thing to note is to ensure the client is closed properly on class destruction or use a context manager